actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,31 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :==, :each, :empty?, to: :@variants
```

Comment:
With globbing:

```ruby
request.variant = [:phone, :tablet]
request.variant == [:phone, :tablet]
# => false
```

Perhaps the solution here is to do something similar to what we do when assigning `request.variant`:

```ruby
def ==(other)
  @variants == Array(other)
end
```